crates/red_knot_python_semantic/src/types/property_tests.rs

```diff
@@ -352,4 +365,18 @@ mod flaky {
         double_negation_is_identity, db,
         forall types t. t.negate(db).negate(db).is_equivalent_to(db, t)
     );
+
+    // ~T should be disjoint from T
+    type_property_test!(
+        negation_is_disjoint, db,
+        forall types t. t.is_fully_static(db) => t.negate(db).is_disjoint_from(db, t)
+    );
+
+    // For fully static types, their intersection should be subtype of them.
+    type_property_test!(
+        fully_static_types_are_supertypes_of_their_intersection, db,
+        forall types s, t.
+            s.is_fully_static(db) && t.is_fully_static(db)
+            => intersection(db, s, t).is_subtype_of(db, s) && intersection(db, s, t).is_subtype_of(db, t)
+    );
```

Comment:
And for non-fully-static types, the intersection should be assignable, I think:

```suggestion
    // For fully static types, their intersection should be subtype of them.
    type_property_test!(
        fully_static_types_are_supertypes_of_their_intersection, db,
        forall types s, t.
            s.is_fully_static(db) && t.is_fully_static(db)
            => intersection(db, s, t).is_subtype_of(db, s) && intersection(db, s, t).is_subtype_of(db, t)
    );

    // And for non-fully-static types, the intersection of a pair of types
    // should be assignable to both types of the pair.
    type_property_test!(
        fully_static_types_are_supertypes_of_their_intersection, db,
        forall types s, t. intersection(db, s, t).is_assignable_to(db, s) && intersection(db, s, t).is_assignable_to(db, t)
    );
```